Libraries/LibWeb/Painting/DisplayListPlayerSkia.cpp

```diff
@@ -1103,6 +1077,41 @@ void DisplayListPlayerSkia::apply_opacity(ApplyOpacity const& command)
     canvas.saveLayer(nullptr, &paint);
 }
 
+void DisplayListPlayerSkia::apply_filters(ApplyFilters const& command)
+{
+    if (command.filter.is_none()) {
```

Comment:
How does this work when `command.opacity < 1` and filters are set? Is the opacity applied twice?

Reply:
Ladybird seems to report opacity correctly when testing with an SVG and with a <div> with some text inside it (same filter applied to both).

The code duplication can create confusion though. What we could do is get rid of ApplyOpacity by itself and merge that into ApplyFilters, or alternatively, get rid of the duplication here. I guess it depends on the scope of what we'd like to keep as a "filter". I would be open to either approach, although my personal preference would be for the second option.

Reply:
note: checking the opacity is less than one could probably be promoted to checking for non-trivial filters being applied (i.e. those introducing a substantial visual change, e.g. `opacity < 1` or `hue ≠ 0°` etc.) as an optimisation. not sure if that is the correct moment to update, or whether it is a good place for an optimisation either.